src/solver/ts-generator/include/antares/solver/ts-generator/prepro.h

```diff
@@ -104,9 +110,27 @@ class PreproThermal
     // max x DAYS_PER_YEAR
     Matrix<> data;
     // Parent thermal cluster
-    std::weak_ptr<const ThermalCluster> itsThermalCluster;
-}; // class PreproThermal
+    YString id;
+    unsigned int unitCount;
+}; // class PreproAvailability
+
+struct LinkTsGeneration
```

Comment:
I still believe that there is a kind of code duplication between this type **LinkTsGeneration** and its thermal cluster counterpart. 
But it also would be not immediate to make it disappear. So I understand if we give up for now.

Reply:
Right